### PR TITLE
OCPBUGSM-4255 Added reset and cancel to eventsHandler

### PR DESCRIPTION
**Pull request by RazRegev**
http://10.46.10.2:8080/job/bm-inventory-subsystem-test/554/console

### CODE REVIEW

File: internal/host/host.go [hunk 1]
```diff
@@ -385,12 +393,20 @@ func (m *Manager) IsRequireUserActionReset(h *models.Host) bool {
 }
 
 func (m *Manager) ResetHost(ctx context.Context, h *models.Host, reason string, db *gorm.DB) *common.ApiErrorResponse {
+	eventSeverity := models.EventSeverityInfo
+	eventInfo := fmt.Sprintf("Updated status of host %s to %s", m.GetHostname(h), models.HostStatusResetting)
```
Comment (avishayt) [inline internal/host/host.go:397]:
```suggestion
	eventInfo := fmt.Sprintf("Reset host installation")
```

File: internal/host/host.go [hunk 2]
```diff
@@ -385,12 +393,20 @@ func (m *Manager) IsRequireUserActionReset(h *models.Host) bool {
...
+	}()
+
 	err := m.sm.Run(TransitionTypeResetHost, newStateHost(h), &TransitionArgsResetHost{
 		ctx:    ctx,
 		reason: reason,
 		db:     db,
 	})
 	if err != nil {
+		eventSeverity = models.EventSeverityError
+		eventInfo = fmt.Sprintf("Failed to reset installation of host %s. Error: %s", m.GetHostname(h), err.Error())
```
Comment (avishayt) [inline internal/host/host.go:409]:
```suggestion
		eventInfo = fmt.Sprintf("Failed to reset host installation. Error: %s", err.Error())
```

File: internal/host/host.go [hunk 3]
```diff
@@ -363,12 +363,20 @@ func (m *Manager) UpdateHostname(ctx context.Context, h *models.Host, hostname s
 }
 
 func (m *Manager) CancelInstallation(ctx context.Context, h *models.Host, reason string, db *gorm.DB) *common.ApiErrorResponse {
+	eventSeverity := models.EventSeverityInfo
+	eventInfo := fmt.Sprintf("Updated status of host %s to %s", m.GetHostname(h), models.HostStatusError)
+	defer func() {
+		m.eventsHandler.AddEvent(ctx, h.ID.String(), eventSeverity, eventInfo, time.Now(), h.ClusterID.String())
```
Comment (avishayt) [inline internal/host/host.go:369]: This should be the host ID

File: internal/host/host.go [hunk 4]
```diff
@@ -385,12 +393,20 @@ func (m *Manager) IsRequireUserActionReset(h *models.Host) bool {
 }
 
 func (m *Manager) ResetHost(ctx context.Context, h *models.Host, reason string, db *gorm.DB) *common.ApiErrorResponse {
+	eventSeverity := models.EventSeverityInfo
+	eventInfo := fmt.Sprintf("Reset host installation")
+	defer func() {
+		m.eventsHandler.AddEvent(ctx, h.ID.String(), eventSeverity, eventInfo, time.Now(), h.ClusterID.String())
```
Comment (avishayt) [inline internal/host/host.go:399]: This should be the host ID

**Comment by avishayt** [comment]
@ronniel1 Do you think there's any reasonable way to add events in middleware for API success and failure?

**Comment by ronniel1** [comment]
> @ronniel1 Do you think there's any reasonable way to add events in middleware for API success and failure?

I guess there is.
You want an event on any API call success/failure?

**Comment by avishayt** [comment]
> > @ronniel1 Do you think there's any reasonable way to add events in middleware for API success and failure?
> 
> I guess there is.
> You want an event on any API call success/failure?

For every non-GET success, and for every failure.  The failure events should include the error, and they should all look like English (probably the hard part).

**Comment by ronniel1** [comment]
> > > @ronniel1 Do you think there's any reasonable way to add events in middleware for API success and failure?
> > 
> > 
> > I guess there is.
> > You want an event on any API call success/failure?
> 
> For every non-GET success, and for every failure. The failure events should include the error, and they should all look like English (probably the hard part).

What about PostNextStepReply?

Do we really want an event for every succesfulll Save&update operation?

**Comment by avishayt** [comment]
> > > > @ronniel1 Do you think there's any reasonable way to add events in middleware for API success and failure?
> > > 
> > > 
> > > I guess there is.
> > > You want an event on any API call success/failure?
> > 
> > 
> > For every non-GET success, and for every failure. The failure events should include the error, and they should all look like English (probably the hard part).
> 
> What about PostNextStepReply?
> 
> Do we really want an event for every succesfulll Save&update operation?

OK, good points on both.  I guess it's easier to just do it manually...

**Pull request merged**